types/types.go

```diff
@@ -226,6 +226,10 @@ type Image interface {
 	// The Digest field is guaranteed to be provided; Size may be -1.
 	// WARNING: The list may contain duplicates, and they are semantically relevant.
 	LayerInfos() []BlobInfo
+	// LayerDiffIDs returns a list of DiffIDs (= digests of the UNCOMPRESSED versions) of layers referenced by this image,
+	// in order (the root layer first, and then successive layered layers), if available, or nil if not.
+	// WARNING: The list may contain duplicates, and they are semantically relevant.
+	LayerDiffIDs() ([]digest.Digest, error)
```

Comment:
`LayerInfos` contains the digest of the downloadable blob, whether or not it is compressed. Anyone wanting to call `ImageSource.GetBlob` needs the `LayerInfos` output.

`LayerDiffIDs` contains the DiffID, which may be useless for getting the layer (i.e. the caller may need to call `LayerInfos`+`GetBlob` and uncompress, or get the data from elsewhere).

I have updated the comments a _tiny_ bit, it could probably still be more explicit. Perhaps the `LayerDiffIDs` documentation should say that “most users should call `LayerInfos` instead”?